### DISCUSSION TITLE
depot_tools RPC failed transiently (retry in 2s):

### DISCUSSION THREAD

**Discussion by Jhevison** [termux/termux-packages#20602] [Q&A] [answered]
Well, for some reason when using proot-distro and trying to use depot_tools, everything works perfectly.  But when I try to do the same thing just using termux, I get this error: 

client.go:312 W] RPC failed transiently (retry in 1s): rpc error: code = Internal desc = prpc: sending request: Post "https ://chrome-infra-packages.appspot.com/prpc/cipd.Repository/ResolveVersion": dial tcp: lookup chrome-infra-packages.appspot.com on [::1]:53: read udp [::1 ]:38251->[::1]:53: read: connection refused {"host":"chrome-infra-packages.appspot.com", "method":"ResolveVersion", "service":"cipd.Repository ", "sleepTime":"1s"}

Yes, I know that this tool is quite buggy, but for some reason, it seems that when trying to connect to the server, it fails, but this does not happen in proot-distro.  Would anyone know why this happens?

**Comment by sylirre** [comment, marked as answer]
The error `dial tcp: lookup chrome-infra-packages.appspot.com on [::1]:53: read udp [::1 ]:38251->[::1]:53: read: connection refused` indicates that program failed to read `/etc/resolv.conf` and attempts fallback to local DNS resolver. This file doesn't exist on Android OS (Termux) at given location.

Termux provides `resolv.conf` at `/data/data/com.termux/files/usr/etc/resolv.conf` when `resolv-conf` package is installed.

Your workarounds are `termux-chroot` or `proot-distro`. Both would bind `resolv.conf` to its normal place (`/etc/resolv.conf`).

Local DNS resolver `[::1]:53` available only when device runs WiFi hotspot.

See https://wiki.termux.com/wiki/Differences_from_Linux (Termux is not FHS compliant)
